File: Samples/CardsStarterKit_4_0_VB/CardsStarterKit_4_0/CardsFramework/Players/Player.vb
#Region "File Description"
'-----------------------------------------------------------------------------
' Player.vb
'
' Microsoft XNA Community Game Platform
' Copyright (C) Microsoft Corporation. All rights reserved.
'-----------------------------------------------------------------------------
#End Region

#Region "Using Statements"
Imports System.Text
#End Region


''' <summary>
''' Represents base player to be extended by inheritance for each
''' card game.
''' </summary>
Public Class Player
#Region "Property"
    Public Property Name As String
    Public Property Game As CardsGame
    Public Property Hand As Hand
#End Region

    Public Sub New(ByVal name As String, ByVal game As CardsGame)
        Me.Name = name
        Me.Game = game
        Hand = New Hand
    End Sub
End Class

File: Samples/Minjie_ARCHIVE_2_0/Minjie/Minjie/Player.cs
#region File Description
//-----------------------------------------------------------------------------
// Player.cs
//
// Microsoft XNA Community Game Platform
// Copyright (C) Microsoft Corporation. All rights reserved.
//-----------------------------------------------------------------------------
#endregion

#region Using Statements
using System;
#endregion

namespace Minjie
{
    /// <summary>
    /// Implementation for all players, regardless of type.
    /// </summary>
    abstract class Player
    {
        private BoardColors boardColor;
        public BoardColors BoardColor
        {
            get { return boardColor; }
        }

        protected Player(BoardColors boardColor)
        {
            if ((boardColor != BoardColors.Black) && (boardColor != BoardColors.White))
            {
                throw new ArgumentException("Invalid board color.");
            }
            this.boardColor = boardColor;
        }

        public abstract void Update(Board board);
    }
}
